### PR TITLE
Improved wording around SSH and key authn with GitHub

### LINKED ISSUE #243: Poor flow of Git authentication and push section
In section 1 episode 4, this section on authentication with Git doesn't read particularly well and seems incomplete:

Screenshot from 2023-10-18 11-41-39

Also, there is no mention of how to set up SSH keys. GitHub has a pretty decent section on this that we should link to: https://docs.github.com/en/authentication/connecting-to-github-with-ssh/adding-a-new-ssh-key-to-your-github-account?tool=webui That documentation page is a bit tricky because it has different versions for Windows, Mac and Linux near the top (but switching between them is by no means obvious).

### PR DESCRIPTION
**Pull request by anenadic**
Improved the wording around SSH and key authentication with GitHub.

1. Stressing more that we recommend SSH for authorised operations with GitHub (e.g. push)
2. Removed discussion around GitHub requiring personal access tokens with HTTPS (old news)
3. Fixed some awkward wording.
4. Used this as an opportunity to fix the git feature diagram to have a more correct timeline for "feature y" branch.

Fixes #243.

### CODE REVIEW

**Review by steve-crouch — APPROVED** [review]

File: _episodes/14-collaboration-using-git.md [hunk 1]
```diff
@@ -259,41 +259,34 @@ $ git pull
...
 {: .language-bash}
 
-> ## Authentication Errors
+In the above command,
+`origin` is an alias for the remote repository you used when cloning the project locally
+(it is called that by convention and set up automatically by Git
+when you run `git clone remote_url` command to replicate a remote repository locally);
+`main` is the name of our main (and currently only) development branch.
+
+> ## GitHub Authentication Error
```
Comment (anenadic) [inline _episodes/14-collaboration-using-git.md:275]:
```suggestion
> ## GitHub Authentication/Authorisation Error
```

File: _episodes/14-collaboration-using-git.md [hunk 2]
```diff
@@ -259,41 +259,34 @@ $ git pull
...
-> then you accidentally cloned the repository using HTTPS and not SSH.
-> You can fix this from the command line by
-> resetting the remote repository URL setting on your local repo:
+> If, at this point (i.e. the first time you try to write to a remote repository on GitHub), 
+> you get a warning/error that HTTPS access is deprecated, or a personal access token is required,
+> then you have cloned the repository using HTTPS and not SSH.
+> You should revisit the [instructions 
+> on setting up your GitHub for SSH and key pair authentication](../setup.html#secure-access-to-github-using-git-from-command-line)
+> and can fix this from the command line by
+> resetting the remote repository URL setting on your local repository:
```
Comment (anenadic) [inline _episodes/14-collaboration-using-git.md:283]:
```suggestion
> changing the remote repository's HTTPS URL to its SSH equivalent:
```

File: _episodes/11-software-project.md [hunk 3]
```diff
@@ -60,8 +60,11 @@ from GitHub within your own GitHub account
 and then obtain a local copy of that project (from your GitHub) on your machine.
 
 1. Make sure you have a GitHub account
-   and that you have set up your SSH key pair for authentication with GitHub,
-   as explained in [Setup](../setup.html#secure-access-to-github-using-git-from-command-line).
+   and that you have set up your **SSH key pair for authentication with GitHub**,
+   as explained in [Setup](../setup.html#secure-access-to-github-using-git-from-command-line). Note 
+   that, while it is possible to use **HTTPS** with a personal access token for authentication with 
+   GitHub, the recommended and supported authentication method to use for this course is via SSH and key 
+   pairs.
```
[the remark this answers is not in the document]
Comment (bielsnohr) [reply _episodes/11-software-project.md:67]: I somehow had missed that we had instructions about SSH key pairs in the Setup instructions!!! These do indeed point to the location that I was suggesting in #265 and therefore resolve most of my suggestions raised in that issue.

File: fig/git-feature-branch.svg
Comment (bielsnohr) [inline fig/git-feature-branch.svg:1]: I personally don't see an issue with the current order of the changes on the Feature Y branch, but it also doesn't hurt to move them where they are now, so I have no problem with this change if it makes explaining things easier when others run the course.

**Comment by anenadic** [comment]
All requested changes done - re-requesting review from @bielsnohr.

**Review by bielsnohr — APPROVED** [review]
Thanks @anenadic . LGTM! Will merge now.

**Pull request merged**